core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -221,11 +295,49 @@ class AdminClient(val time: Time,
   }
 
   def close() {
+    running = false
```

Comment:
Are you suggesting to wait for NetworkThread to exit before or after calling `client.close()`? If we call `client.close()` first, we can not prevent error from being logged because `Selector.close()` will log error. If we wait for NetworkThread to exit first, then close() may become a blocking call which seems unnecessary. 

I agree with you that we do not want to throw Exception here. But I am not sure why we should not log any error here. For example, `Selector.close()` may log error if there is `IOException`.

I have updated the code to catch exception. Does this address the problem?